modular_splurt/code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -74,3 +74,86 @@
 		var/temp = holder ? holder.chem_temp : T20C
 		T.atmos_spawn_air("copium=[volume];TEMP=[temp]")
 	return
+
+
+/datum/reagent/drug/genital_purity
+	name = "Nope"
+	taste_description = "purity"
+	metabolization_rate = 0.2
+	var/mod_flag = GENITAL_IMPOTENT
+	var/genital
+
+/datum/reagent/drug/genital_purity/on_mob_metabolize(mob/living/carbon/C)
+	..()
+	var/obj/item/organ/genital/G = C.getorganslot(genital)
+	if(!G)
+		return
+	
+	G.set_aroused_state(0, "impotence") //just so it goes down if it's a penis
+	ENABLE_BITFIELD(G.genital_flags, mod_flag)
+
+/datum/reagent/drug/genital_purity/on_mob_end_metabolize(mob/living/carbon/C)
+	var/obj/item/organ/genital/G = C.getorganslot(genital)
+	if(!genital)
+		return
+	
+	DISABLE_BITFIELD(G.genital_flags, mod_flag)
+
+	..()
+
+/datum/reagent/drug/genital_purity/penis_purity
+	name = "Penis Purity"
+	color = COLOR_STRONG_VIOLET
+	genital = ORGAN_SLOT_PENIS
+
+/datum/reagent/drug/genital_purity/virtuous_vagina
+	name = "Virtuous Vagina"
+	color = COLOR_LIGHT_PINK
+	genital = ORGAN_SLOT_VAGINA
+
+/datum/reagent/drug/genital_stimulator
+	name = "Nope"
+	metabolization_rate = 0.2
+	taste_description = "excitement"
+	var/mod_flag = GENITAL_OVERSTIM
+	var/genital
+
+/datum/reagent/drug/genital_stimulator/on_mob_metabolize(mob/living/carbon/C)
+	..()
+	if(genital == "anus")
+		ADD_TRAIT(C, mod_flag, ORGAN_TRAIT)
+		return
+
+	var/obj/item/organ/genital/G = C.getorganslot(genital)
+	if(!G)
+		return
+	else
+		G.genital_flags |= mod_flag
+
+/datum/reagent/drug/genital_stimulator/on_mob_end_metabolize(mob/living/carbon/C)
+	var/obj/item/organ/genital/G = C.getorganslot(genital)
+	if(!genital)
+		return
+	
+	if(G == "anus")
+		REMOVE_TRAIT(C, mod_flag, ORGAN_TRAIT)
+	else
+		DISABLE_BITFIELD(G.genital_flags, mod_flag)
+
+	..()
+
+/datum/reagent/drug/genital_stimulator/anal_allure
+	name = "Anal Allure"
+	color = COLOR_DARK_RED
+	mod_flag = TRAIT_OVERSTIM_ANUS
+	genital = "anus"
+
+/datum/reagent/drug/genital_stimulator/breast_buzzer
+	name = "Breast Buzzer"
+	color = COLOR_PINK
+	genital = ORGAN_SLOT_BREASTS
+
+/datum/reagent/drug/genital_stimulator/peen_pop
+	name = "Peen Pop"
+	color = COLOR_MOSTLY_PURE_ORANGE
+	genital = ORGAN_SLOT_PENIS
```

Comment:
You'll wanna make sure that these reagents first check for the appropriate pref toggles to be on depending on the flags they apply before they have effects on a character. Setting up said prefs as I mentioned before